gapic-generator/templates/default/rakefile.erb

```diff
@@ -0,0 +1,138 @@
+<%- assert_locals gem -%>
+<%= render partial: "shared/license" %>
+
+require "bundler/setup"
+require "bundler/gem_tasks"
+
+require "rubocop/rake_task"
+RuboCop::RakeTask.new
+
+require "rake/testtask"
+desc "Run tests."
+Rake::TestTask.new do |t|
+  t.test_files = FileList["test/**/*_test.rb"]
+  t.warning = false
+end
+
+namespace :test do
+  desc "Runs tests with coverage."
+  task :coverage do
+    require "simplecov"
+    SimpleCov.start do
+      command_name "<%= gem.name %>"
+      track_files "lib/**/*.rb"
+      add_filter "test/"
+    end
+
+    Rake::Task[:test].invoke
+  end
+end
+
+desc "Runs the smoke tests."
+desc "Runs the smoke tests."
+Rake::TestTask.new :smoke_test do |t|
+  t.test_files = FileList["acceptance/**/*smoke_test.rb"]
+  t.warning = false
+end
+
+namespace :smoke_test do
+  desc "Runs smoke tests with coverage."
+  task :coverage do
+    require "simplecov"
+    SimpleCov.start do
+      command_name "<%= gem.name %>"
+      track_files "lib/**/*.rb"
+      add_filter "test/"
+    end
+
+    Rake::Task[:smoke_test].invoke
+  end
+end
+
+# Acceptance tests
+desc "Run the <%= gem.name %> acceptance tests."
+task :acceptance, :project, :keyfile do |t, args|
```

Comment:
Do you want acceptance tests in the default generator (base)? Do you want to test against live services by default? Or do you want the rake task definition to be different?

Reply:
We should run whatever tests we can reasonably generate. This part of the Rakefile deals with keyfiles and project names, which I don't think generalize to non-Google projects - do they?

Reply:
The default generator creates a gemspec file, so this seems reasonable to me.

Reply:
What's the relationship between the gemspec file and the `:project, :keyfile` args here?

Reply:
My point is that this isn't cloud-specific, it is gem-specific. If the default generator is outputting a gem, with guidance on authenticating via the credentials class that each service in the gem has. The credentials even uses the `env_var_prefix` value for the gem, either calculated or configured. If we have all that in the gem, why not also have a rake task to run acceptance tests?

Reply:
I'm still not sure about this, but I think we can defer it for now since it's not really that important of an issue to solve now. Auth should be different in both cases and it's not clear to me that the concepts of a `keyfile` or `project`, etc. should be used in the general case. 

For now, I'd suggest that we leave it as-is if the tests will actually work and remove it if not. We can come back to how auth should work outside of Cloud later.

Reply:
Oh, the project argument is not part of credentials and that is cloud-specific. I'll just remove them all then.